### PR TITLE
Migrated email filter columns from VARCHAR to TEXT

### PR DESCRIPTION
**Pull request by allouis**
refs TryGhost/Team#1287

Currently we have a hard limit of how large an email filter can be,
which is very restrictive once a site starts using Tiers - by moving
toward a TEXT column, we essentially give the filters unlimited size.

This currently doesn't handle SQLite as I'm waiting on feedback from
Matt on whether or not the development database needs this migration.

### CODE REVIEW

File: core/server/data/migrations/versions/4.35/2022-02-01-11-48-update-email-recipient-filter-column-type.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        });
+    },
+    async function down(knex) {
+        if (knex.client.config.client === 'sqlite3') {
+            logging.warn('Skipping migration for SQLite3');
+            return;
+        }
+        logging.info('Changing posts.email_recipient_filter column from TEXT to VARCHAR(50)');
+        knex.schema.alterTable('posts', function (table) {
+            table.string('email_recipient_filter', 50).alter();
```
Comment (matthanley) [inline core/server/data/migrations/versions/4.35/2022-02-01-11-48-update-email-recipient-filter-column-type.js:22]: These `down()` operations will fail if content is longer than 50 chars:
```mysql
mysql> alter table posts modify email_recipient_filter varchar(50);
ERROR 1406 (22001): Data too long for column 'email_recipient_filter' at row 1
```

**Review by matthanley — CHANGES_REQUESTED** [review]

File: core/server/data/migrations/versions/4.35/2022-02-01-11-48-update-email-recipient-filter-column-type.js (hunk 1 above)
Comment (allouis) [reply core/server/data/migrations/versions/4.35/2022-02-01-11-48-update-email-recipient-filter-column-type.js:22]: Smart! What shall we do? noop on `down` I guess?

Comment (allouis) [reply core/server/data/migrations/versions/4.35/2022-02-01-11-48-update-email-recipient-filter-column-type.js]: Done

**Review by rshbhgrg — APPROVED** [review]

**Comment by daniellockyer** [comment]
@allouis Can I check that we'll be adding the default to the model? Changing to a `text` removes the default on the column

**Comment by allouis** [comment]
@daniellockyer Good question! Both models have a default set https://github.com/TryGhost/Ghost/blob/main/core/server/models/post.js#L68 https://github.com/TryGhost/Ghost/blob/main/core/server/models/email.js#L11

**Review by daniellockyer — APPROVED** [review]

**Comment by daniellockyer** [comment]
I'm happy with this with the understanding we aren't providing a `down` to revert the column change to keep things simpler

**Pull request merged**